### ISSUE TITLE
bug?: retry not working

### ISSUE THREAD

**Issue by roachadam** [softonic/axios-retry#161]
Not quite sure how this happened, but I'm probably missing something stupid.

Code:
```js
axiosRetry(this.ax, { 
	retries: 3,
	retryCondition: (e) => {
		//return true;
		if(!e.response)
			return false;

		//todo 302
		return e.response.status >=200 && e.response.status <=204;
	}
});
```
For whatever reason, I make a request with that same ax instance, get a 502 error, and it does not retry. I also uncommented the return true, and it still does not retry.

I thought it could be interfering with my other axios interceptor, but I turned that off and still the same issue persists.

win64 bit
node v12.16.1
axios-retry v

**Comment by ffxsam** [comment]
Same here, non-working right out of the box. Pretty basic test, and I never see my `console.log` statement:

```js
const axios = require('axios');
const axiosRetry = require('axios-retry');

axiosRetry(axios, {
  retries: 3,
  retryDelay: retryCount => {
    console.log('retry count:', retryCount);
    return retryCount * 1000;
  }
});

(async function() {
  const data = await axios.get('http://httpstat.us/503');
})();
```

**Comment by schmidlidev** [comment]
This library seems to just... not work? I even tried:

> retryCondition: () => true

and could not get this library to ever perform a retry.
Ended up using https://www.npmjs.com/package/retry-axios instead.

**Comment by geoffrich** [comment]
I ran into this issue, and for me it was caused by my Axios version. 0.21.2 introduced a bug that [0.21.3](https://github.com/axios/axios/blob/master/CHANGELOG.md#0213-september-4-2021) fixed. Updating Axios caused the retries to succeed.

That likely isn't the cause of OP's issue, since they created this issue in February and 0.21.2 was released in September, but wanted to comment in case someone else had the same root cause.
